### DISCUSSION TITLE
Create local host directories for volume bind mount implicitly?

### DISCUSSION THREAD

**Discussion by polarathene** [containers/podman#24386] [Q&A] [answered]
I have been getting familiar with Quadlets. With Docker this is handled implicitly for `--volume` / `-v` bind mounts or equivalent `volumes`/`configs` in `compose.yaml`.

When providing a sample/reference Quadlet for your project or documentation, should the user be expected to create each local volume path themselves? Or would you add that into the Quadlet itself with something like this:

```ini
# .config/containers/systemd/my-service.container 
[Container]
# Map root ownership to rootless host user:
# - Triggers a `chown` copy of the container image content,
#   in addition to mapping ownership for the volume.
# - GIDMap defaults to the same ID mapping as UIDMap.
UIDMap=+0:@%U

# Mount `~/volumes/my-service/some-dir` with SELinux compatibility:
Volume=%h/volumes/%N/some-dir:/some-dir:Z

[Service]
# Create the `Volume` location before starting the container:
ExecStartPre=mkdir -p %h/volumes/%N/some-dir
```

---

With `compose.yaml` I am familiar with being able to declare this file anywhere and run it, so for reproductions a relative host path for each volume can be run with a `compose.yaml` at `/tmp` for example, or the `compose.yaml` is part of a git project that can be cloned and run from there.

With Quadlets, these are always managed in common locations rather than ad-hoc? Would you distribute them similar to `compose.yaml` in a git project? I assume using them from the git clone location would be a bit awkward if possible.

---

Just checking what the expectation is from the podman community side. Are users more likely to manually copy the equivalent Quadlet file(s) to the standard locations Quadlet uses, and thus also check each Volume definition to make these bind mounts?

I am aware of named data volumes with `.volume` files, this question is more oriented towards how projects should support/document the local bind mount volumes scenario.

**Comment by eriksjolund** [comment, marked as answer]
Using `ExecStartPre=` to create a directory sounds good to me. 
Just like you suggested above:
```
[Service]
# Create the `Volume` location before starting the container:
ExecStartPre=mkdir -p %h/volumes/%N/some-dir
```
